interp/src/interpreter/interpret_group.rs

```diff
@@ -165,69 +160,90 @@ fn interp_assignments<'a, I: Iterator<Item = &'a ir::Assignment>>(
 
     let cells = get_cells(assigns.iter().copied());
 
+    let possible_ports: HashSet<*const ir::Port> =
+        assigns.iter().map(|a| get_const_from_rrc(&a.dst)).collect();
     let mut val_changed_flag = false;
 
     while !is_signal_high(working_env.get(done_signal)) || val_changed_flag {
+        //helps us tell if there are multiple assignments to same port >:0
+        let mut assigned_ports: HashSet<*const ir::Port> = HashSet::new();
```

Comment:
Silly question that probably has an obvious answer: is there a reason why this tracking isn't done already by `working_env`? That is, it seems like the "top layer" in the smoosher would know what ports had already been assigned to…

Reply:
Perhaps the answer to my question is that updated values go into `updates_list`, not into `working_env`. Maybe I'm missing something obvious, but wouldn't this be the perfect place to use the capabilities of the Smoosher? A fork could buffer up all the updates, replacing both `updates_list` and `assigned_ports` and removing the need for a "manual" smooshing loop to apply updates.

Reply:
That got a bit rambly oops. The basic answer is yes! But the types of the different environments will get in the way unless we duplicate some values.

Reply:
That helps! The need for different value types clarifies why the solution isn't just "everything should just be a Smoosher." Maybe exploiting the Smoosher to simplify things narrowly here is best done in (eventual, not prioritized) follow-on work and not in this PR…